app/react/Library/components/TableRow.tsx

```diff
@@ -0,0 +1,170 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+
+import { PropertySchema } from 'shared/types/commonTypes';
+import { TemplateSchema } from 'shared/types/templateType';
+import { ThesaurusSchema } from 'shared/types/thesaurusType';
+import formatter from 'app/Metadata/helpers/formater';
+import MarkdownViewer from 'app/Markdown';
+import { I18NLink } from 'app/I18N';
+import GeolocationViewer from 'app/Metadata/components/GeolocationViewer';
+
+interface TableRowProps {
+  columns: PropertySchema[];
+  document: any;
+  storeKey: 'library' | 'uploads';
+  selected?: boolean;
+  onClick: any;
+  templates: any;
+  thesauris: any;
+  zoomLevel: number;
+}
+
+function getLink(url: string, label: string) {
+  return (
+    <I18NLink key={url} to={url}>
+      {label}
+    </I18NLink>
+  );
+}
+
+function formatProperty(prop: any) {
+  let result = prop?.value;
+  if (!result) {
+    return typeof prop === 'string' ? prop : undefined;
+  }
+  if (['date', 'daterange', 'numeric', 'select', 'text', undefined].includes(prop.type)) {
+    return result;
+  }
+
+  switch (prop.type) {
+    case 'multiselect':
+    case 'multidaterange':
+    case 'multidate':
+      result = prop.value.map((p: any) => p.value).join(', ');
+      break;
+    case 'markdown':
+      result = <MarkdownViewer markdown={prop.value} />;
+      break;
+    case 'link':
+      result = getLink(prop.value.url, prop.value.label);
+      break;
+    case 'relationship':
+      result = prop.value.map((p: any, index: number) => (
+        <React.Fragment key={index}>
+          {index > 0 && ', '}
+          {getLink(p.url, p.value)}
+        </React.Fragment>
+      ));
+      break;
+    case 'geolocation':
+      result = <GeolocationViewer points={prop.value} onlyForCards />;
+      break;
+    default:
+      result = undefined;
+      break;
+  }
+  return result;
+}
+
+function formatDocument(document: any, templates: TemplateSchema[], thesauris: ThesaurusSchema[]) {
+  let formattedDoc = document.toJS();
+  const template = templates.find((t: TemplateSchema) => t.get('_id') === formattedDoc.template);
+  formattedDoc.templateName = template?.get('name');
```

Comment:
We already have a component that renders template names based on template ID.  Is this really necessary here or could we use that component instead?

Reply:
Given that the function that we are using to format the entity already has contemplated putting the name of the template, we fixed the assignation and use that property instead. https://github.com/huridocs/uwazi/blob/5bed5193603aaab021a925248248c7e10f814399/app/react/Metadata/helpers/formater.js#L322